### PR TITLE
Opencv fix compile for android

### PR DESCRIPTION
**Pull request by bog-dan-ro**
Specify library name and version:  **lib/1.0**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/opencv/4.x/conanfile.py [hunk 1]
```diff
@@ -504,7 +504,8 @@ def _configure_cmake(self):
                 "armv8": "aarch64",
                 "armv8.3": "aarch64",
             }.get(str(self.settings.arch), str(self.settings.arch))
-            self._cmake.definitions["CMAKE_SYSTEM_PROCESSOR"] = cmake_system_processor
+            if self.settings.os != "Android":
+                self._cmake.definitions["CMAKE_SYSTEM_PROCESSOR"] = cmake_system_processor
```
Comment (SpaceIm) [inline recipes/opencv/4.x/conanfile.py:508]:
```suggestion
            self._cmake.definitions["CONAN_OPENCV_SYSTEM_PROCESSOR"] = cmake_system_processor
```

and add this in CMakeLists wrapper after `conan_basic_setup`:

```cmake
if(NOT CMAKE_SYSTEM_PROCESSOR)
    set(CMAKE_SYSTEM_PROCESSOR ${CONAN_OPENCV_SYSTEM_PROCESSOR})
endif()
```

File: recipes/opencv/4.x/conanfile.py [hunk 2]
```diff
@@ -220,7 +220,7 @@ def validate(self):
         if self.options.shared and self._is_msvc and "MT" in msvc_runtime_flag(self):
             raise ConanInvalidConfiguration("Visual Studio with static runtime is not supported for shared library.")
         if self.settings.compiler == "clang" and tools.Version(self.settings.compiler.version) < "4":
-            raise ConanInvalidConfiguration("Clang 3.x can build OpenCV 4.x due an internal bug.")
+            raise ConanInvalidConfiguration("Clang 3.x can't build OpenCV 4.x due an internal bug.")
```
Comment (SpaceIm) [inline recipes/opencv/4.x/conanfile.py:223]:
```suggestion
            raise ConanInvalidConfiguration("Clang 3.x can't build OpenCV 4.x due to an internal bug.")
```

**Review by SpaceIm — APPROVED** [review]

**Comment by SpaceIm** [comment]
@bog-dan-ro please request access to EAP in https://github.com/conan-io/conan-center-index/issues/4

**Comment by bog-dan-ro** [comment]
> @bog-dan-ro please request EAP in #4

What EAP stands for ?

**Comment by SpaceIm** [comment]
Early Access Program

**Comment by bog-dan-ro** [comment]
@SpaceIm  I got the EAP but I'm afraid I don't know what to do with it :).
If possible please merge this patch for me.
Thanks.

**Comment by SpaceIm** [comment]
Just Close/open this PR to trigger CI again.

**Review by uilianries — APPROVED** [review]

**Pull request merged**